openpype/hosts/nuke/api/plugin.py

```diff
@@ -818,15 +817,13 @@ def __init__(self,
 
         self.file = self.fhead + self.name + ".{}".format(self.ext)
         if ".{}".format(self.ext) not in VIDEO_EXTENSIONS:
-            # filename would be with frame hashes if
-            # the file extension is not in video format
-            filename = get_head_filename_without_hashes(
-                self.path_in, self.name)
-            self.file = filename
+            filename = os.path.basename(self.path_in)
+            self.file = re.sub(
+                self.fhead, self.fhead + self.name + ".", filename)
```

Comment:
`self.fhead` does currently not include the frame number or frame hashing as far as I could tell from the logic in [`get_file_info`](https://github.com/ynput/OpenPype/blob/13b46070fe9fef0b6474a50e9cebb7d7b73eac43/openpype/hosts/nuke/api/plugin.py#L584-L608) - is that what you intended to produce here?

Also, if you use regex on a name/pattern you want to exactly match use `re.escape()` on the input because otherwise e.g. `hello.world` would also match `hello_world` or `hellooworld` since the `.` means any character. 

Could you explain what, in your eyes the different variables should be?
```
self.path_in = "/path/to/hello.####.exr"
self.fhead = "hello."
self.name = "baking"
filename = `hello.####.exr`
```
Does this seem about right?

If so, this is likely your best shot to 'keeping whatever is in between' head and extension.
```python
# Strip off the head but preserve as much of the remainder of the filename
filename = filename[len(self.fhead):]
```
Better:
```python
filename_no_ext = os.path.splitext(os.path.basename(self.path_in))[0]
after_head = filename_no_ext[len(self.fhead):]
self.file = self.fhead + self.name + after_head + self.ext
```

And this logic I suspect might even hold true for both video files and image files, for video files `after_head` would just be an empty string.

This is just quickly out of the top of my head.